File: java/samples/batch-mode/src/main/java/com/myscript/iink/samples/batchmode/Stroke.java
/*
 * Copyright (c) MyScript. All rights reserved.
 */

package com.myscript.iink.samples.batchmode;

import com.myscript.iink.PointerType;

import java.util.Arrays;

public class Stroke {
    private PointerType pointerType;
    private int pointerId;
    private float[] x;
    private float[] y;
    private long[] t;
    private float[] p;

    PointerType getPointerType() {
        return pointerType;
    }

    void setPointerType(PointerType pointerType) {
        this.pointerType = pointerType;
    }

    int getPointerId() {
        return pointerId;
    }

    void setPointerId(int pointerId) {
        this.pointerId = pointerId;
    }

    float[] getX() {
        return x;
    }

    void setX(float[] x) {
        this.x = x;
    }

    float[] getY() {
        return y;
    }

    void setY(float[] y) {
        this.y = y;
    }

    long[] getT() {
        return t;
    }

    void setT(long[] t) {
        this.t = t;
    }

    float[] getP() {
        return p;
    }

    void setP(float[] p) {
        this.p = p;
    }

    @Override
    public String toString() {
        String result = "{";

        result += "\"pointerType\":" + "\"" + this.getPointerType() + "\",";
        result += "\"pointerId\":" + this.getPointerId() + ",";
        result += "\"x\":" + Arrays.toString(this.getX()) + ",";
        result += "\"y\":" + Arrays.toString(this.getY()) + ",";
        result += "\"t\":" + Arrays.toString(this.getT()) + ",";
        result += "\"p\":" + Arrays.toString(this.getP());

        result += "}";

        return result;
    }
}

File: samples/batch-mode/src/main/java/com/myscript/iink/samples/batchmode/Stroke.kt
/*
 * Copyright (c) MyScript. All rights reserved.
 */
package com.myscript.iink.samples.batchmode

import com.myscript.iink.PointerType
import java.util.*

data class Stroke(
    val pointerType: PointerType,
    val pointerId: Int = 0,
    val x: FloatArray,
    val y: FloatArray,
    val t: LongArray,
    val p: FloatArray
) {
    override fun toString(): String = "{" +
            "\"pointerType\":\"$pointerType\"," +
            "\"pointerId\":$pointerId," +
            "\"x\":${x.contentToString()}," +
            "\"y\":${y.contentToString()}," +
            "\"t\":${t.contentToString()}," +
            "\"p\":${p.contentToString()}" +
            "}"
}